src/@chakra-ui/gatsby-plugin/components/Heading.ts

```diff
@@ -0,0 +1,39 @@
+import { defineStyle, defineStyleConfig } from "@chakra-ui/react"
+import { headingDefaultTheme } from "./components.utils"
+
+const { sizes: defaultSizes } = headingDefaultTheme
+
+const lineHeightScale = {
+  "4xl": "6xs",
+  "3xl": ["5xs", null, "6xs"],
+  "2xl": ["5xs", null, "4xs"],
+  xl: ["3xs", null, "2xs"],
+  lg: ["3xs", null, "2xs"],
+  md: "xs",
+  sm: "base",
+  xs: "base",
+}
+
+/*
+ * Instead of rewriting the entire sizes object, take the existing value from the
+ * default theme and replace the lineHeight values.
+ */
+const sizes = Object.entries(defaultSizes || {}).reduceRight(
+  (acc, [key, value]) => {
+    return {
+      ...acc,
+      [key]: defineStyle({
+        ...value,
+        lineHeight: lineHeightScale[key],
```

Comment:
Trying to understand how this mapping works @TylerAPfledderer 

On the Chakra default theme I see these possible values for `lineHeight`

```
lineHeights: {
    normal: "normal",
    none: 1,
    shorter: 1.25,
    short: 1.375,
    base: 1.5,
    tall: 1.625,
    taller: "2",
    "3": ".75rem",
    "4": "1rem",
    "5": "1.25rem",
    "6": "1.5rem",
    "7": "1.75rem",
    "8": "2rem",
    "9": "2.25rem",
    "10": "2.5rem",
  }
```

but in the mapping defined previously, we are using for example, `"xs"` for the `md` size. Where is this `xs` coming from in this case?

Reply:
Ah! I see now that we have defined them on the `typography.ts` file, got it now! :)